src/queries/subjects.ts

```diff
@@ -1,13 +1,35 @@
+import { useState, useEffect } from "react";
+
 export interface Subject {
   name: string;
   lastDate: string;
 }
 
 export function useSubjects(): Subject[] | null {
-  const subjects = [
-    { name: "PBL演習", lastDate: "2023-01-01" },
-    { name: "サイバーセキュリティ", lastDate: "2023-01-02" },
-    // その他の科目...
-  ];
+  const [subjects, setSubjects] = useState<Subject[] | null>(null);
+
+  useEffect(() => {
+    const controller = new AbortController();
+    const signal = controller.signal;
+
+    const SUBJECTS_ENDPOINT = "https://backend.mikuroxina.workers.dev/subjects";
+    (async () => {
+      try {
+        const response = await fetch(SUBJECTS_ENDPOINT, { signal });
+        if (!response.ok) {
+          setSubjects(null);
+          return;
+        }
+        const fetchedSubjects: Subject[] = await response.json();
+        setSubjects(fetchedSubjects);
+      } catch {
+        setSubjects(null);
```

Comment:
```suggestion
          return;
        }
        const fetchedSubjects: Subject[] = await response.json();
        setSubjects(fetchedSubjects);
      } catch {
```
エラーの値を設定するならまだしも, `setSubjects(null)` による `null` の設定は不要と思います.